test/annexB/built-ins/String/prototype/replaceAll/custom-replacer-emulates-undefined.js

```diff
@@ -18,7 +18,7 @@ features: [Symbol.replace, String.prototype.replaceAll, IsHTMLDDA]
 
 var searchValue = $262.IsHTMLDDA;
 var replacerGets = 0;
-Object.defineProperty(searchValue, Symbol.replaceAll, {
+Object.defineProperty(searchValue, Symbol.replace, {
   get: function() {
     replacerGets += 1;
     return searchValue;
```

Comment:
this test still lacks the part where the returned value here gets called. 

`2.d.i. Return ? Call(replacer, searchValue, « O, replaceValue »).`

It's good to check for the GetMethod on `@@replace`, but it would be interesting to confirm it calls it with the given args and reuses the given return.

```suggestion
    return function(zero, value) {
      // assert zero
      // assert value
      // assert this
      replacerCalls += 1; // this should be asserted along with replacerGets
    };
```

Reply:
We need to return `[[IsHTMLDDA]]` object here to make sure it is not treated as `null` by `GetMethod`.

JSC code (GetMethod is inlined):

```js
if (!@isUndefinedOrNull(search)) {
    if (@isRegExp(search) && !@stringIncludesInternal.@call(@toString(search.flags), "g"))
        @throwTypeError("String.prototype.replaceAll argument must not be a non-global regular expression");

    var replacer = search.@@replace;
    if (!@isUndefinedOrNull(replacer)) {
        return replacer.@call(search, this, replace);
    }
}
```

We do compare return value to `null`, and I am sure there is coverage for call arguments.

Reply:
> It's good to check for the GetMethod on @@replace, but it would be interesting to confirm it calls it with the given args and reuses the given return.

I suggest letting @shvaikalesh address changes that are not IsHTMLDDA specific in a follow up PR